src/cryptonote_core/blockchain.cpp

```diff
@@ -4396,6 +4399,14 @@ bool Blockchain::add_new_block(const block& bl, block_verification_context& bvc)
 
   rtxn_guard.stop();
   return handle_block_to_main_chain(bl, id, bvc);
+
+  }
+  catch (const std::exception &e)
+  {
+    LOG_ERROR("Exception at [add_new_block], what=" << e.what());
```

Comment:
Unfortunately `LOG_ERROR` might throw as well. If we are ok with it, we could add exceptions logging to `epee: guard against exceptions in RPC handlers` try catch blocks.

Reply:
True, but that's like I/O error and OOM, local things. If you remove logging, you still have things like assigning a string literal to a std::string, which can throw too (OOM), and the packing of the structure to json, and the I/O to send the error response back to the client.

Reply:
Sounds like it might be worth it to add `e.what()` logging to https://github.com/monero-project/monero/pull/6727/files#diff-f6bde7308082ff9a0f25bf8800332d06R113 and similar places.